### DISCUSSION TITLE
Render template on CTRL + S

### DISCUSSION THREAD

**Discussion by psociety** [bootstrap-email/bootstrap-email#153] [Ideas]
Pleeease, pleeeease.
I can't avoid pressing CTRL + S everytime i want to save the changes at https://app.bootstrapemail.com

```js
document.addEventListener("keydown", (e) => {
    if (e.key == 's' && (navigator.platform.match("Mac") ? e.metaKey : e.ctrlKey)) {
        e.preventDefault();
        // SAVE
    }
}, false);
```

**Comment by stuyam** [comment]
Love it, great idea. Adding keyboard shortcuts in general is a great idea. I'm thinking:
- when you hit CTRL + S and the email has not saved it will save it for the first time
- when you hit CTRL + S and the email is already saved it will essentially do a "Rerender" which saves

But what about a keyboard shortcut to just Render? Would CTRL + R be too weird since that is refresh in browsers?

**Reply by psociety** [reply]
I believe that taking over CTRL + R will make you go crazy/rage when developing the service, as it's a key that one usually needs to refresh the page during the development/debugging.

**Reply by stuyam** [reply]
What would be a good alternative? CTRL + E as in "rEnder" or maybe CTRL + D as in "renDer" and the D is right next to the S for save?

**Reply by psociety** [reply]
CTRL + E seems a good fit for. CTRL + D is for bookmarking which i believe it's something you don't want to block off the user.

**Reply by stuyam** [reply]
Haven't forgotten about this. Working on it...
